doc/VEHICLES_CODE.md

```diff
@@ -0,0 +1,74 @@
+# Understanding the vehicles code
+
+## Major structures
+
+### vehicle class
+Each vehicle is stored in an instance of the vehicle class.  The vehicle class contains tilerays for facing and movement, integers for current speed and cruise speed setpoint in hundredths of miles per hour, and a vector of vehicle_parts.
+
+### vehicle_part class
+Every component of a vehicle is stored in an instance of the vehicle_part class.  Each one has a reference to its vpart_info, a point mount that has the x, y coordinates of the component in the vehicle's internal grid, a 2 member array of points named precalc which has the pretranslated values of mount to the vehicle's face.dir and turn.dir, and a item instance named base which is the item used to install the part.
+
+### vpart_info class
+vpart_info stores the JSON data for each type of vehicle component.
+
+## Code flow
+Each turn, vehicles are updated in the following sequence:
```

Comment:
Which is true until somebody changes it in the source and won't update it here because they don't know you copied it here.

Reply:
Yes, which is entirely why I wrote that for my private reference and haven't included it any of my code pushes until I stupidly got lazy today and did `git add doc`.  Removed.